packages/core/src/browser/core-preferences.ts

```diff
@@ -225,6 +225,19 @@ export const corePreferenceSchema: PreferenceSchema = {
             default: isOSX ? 1500 : 500,
             description: nls.localizeByDefault('Controls the delay in milliseconds after which the hover is shown.')
         },
+        'workbench.startupEditor': {
+            type: 'string',
+            enum: ['none', 'welcomePage', 'readme', 'newUntitledFile', 'welcomePageInEmptyWorkbench'],
+            enumDescriptions: [
+                nls.localizeByDefault('Start without an editor.'),
+                nls.localize('theia/getting-started/startup-editor/welcomePage', 'Open the Welcome page, with content to aid in getting started with {0} and extensions.', FrontendApplicationConfigProvider.get().applicationName),
+                nls.localizeByDefault('Open the README when opening a folder that contains one, fallback to \'welcomePage\' otherwise. Note: This is only observed as a global configuration, it will be ignored if set in a workspace or folder configuration.'),
+                nls.localizeByDefault('Open a new untitled text file (only applies when opening an empty window).'),
+                nls.localizeByDefault('Open the Welcome page when opening an empty workbench.'),
+            ],
+            default: 'welcomePage',
+            description: nls.localizeByDefault('Controls which editor is shown at startup, if none are restored from the previous session.')
+        },
```

Comment:
Given that the implementation is completely present in `@theia/getting-started` I believe the preference should be moved there. With the current approach the preference would not be usable if the application does not include `@theia/getting-started`.